src/ssl.c

```diff
@@ -36426,7 +36490,89 @@ WOLFSSL_STACK* wolfSSL_PKCS7_get0_signers(PKCS7* pkcs7, WOLFSSL_STACK* certs,
 
     return signers;
 }
-#endif
+
+/******************************************************************************
+* wolfSSL_PEM_write_bio_PKCS7 - writes the PKCS7 data to BIO
+*
+* RETURNS:
+* returns WOLFSSL_SUCCESS on success, otherwise returns WOLFSSL_FAILURE
+*/
+
+int wolfSSL_PEM_write_bio_PKCS7(WOLFSSL_BIO* bio, PKCS7* p7)
+{
+    WOLFSSL_ENTER("wolfSSL_PEM_write_bio_PKCS7()");
+    byte        outputHead[2048];
+    byte        outputFoot[2048];
+    word32      outputHeadSz = (word32)sizeof(outputHead);
+    word32      outputFootSz = (word32)sizeof(outputFoot);
+    word32 outputSz = 0;
+    byte*  output = NULL;
+    byte*  pem = NULL;
+    int    pemSz = -1;
+    enum wc_HashType hashType;
+    byte hashBuf[WC_MAX_DIGEST_SIZE];
+    word32 hashSz = -1;
+
+    if (bio == NULL || p7 == NULL)
+        return WOLFSSL_FAILURE;
+
+    hashType = wc_OidGetHash(p7->hashOID);
+    hashSz = wc_HashGetDigestSize(hashType);
+    if (hashSz > WC_MAX_DIGEST_SIZE)
+        return WOLFSSL_FAILURE;
+
+    /* only SIGNED_DATA is supported */
+    switch (p7->contentOID) {
+        case SIGNED_DATA:
+            break;
+        default:
+            WOLFSSL_MSG("Unknown PKCS#7 Type");
+            return WOLFSSL_FAILURE;
+    };
+
+    wc_PKCS7_EncodeSignedData_ex(p7, hashBuf, hashSz,
+                        outputHead, &outputHeadSz, outputFoot, &outputFootSz);
+
+    outputSz = outputHeadSz + p7->contentSz + outputFootSz;
+    output = (byte*)XMALLOC(outputSz, NULL, DYNAMIC_TYPE_TMP_BUFFER);
```

Comment:
WOLFSSL_BIO and PKCS7 have heap hints to use when allocating memory. Please use one of them instead of NULL when possible for XMALLOC/XFREE. This will help the staticmemory build and avoid unnecessary technical debt.